### Display Sample Data

In [6]:
%run '../lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


In [7]:
dfFlt = load_df('dfFlt')
cats = get_var_cats()

#### Display Sample Data

In [14]:
dfFlt.select(*cats['orgVars']).distinct().toPandas()

org_id
0  349115
1  360421
2  360541

- 3 organizations

In [15]:
dfFlt.filter( F.col('org_id') == 360421).select(*cats['sectionVars']).distinct().toPandas()

section_id
0       498961
1       499023
2       537231
3       498945
4       498977
5       498971
6       498965
7       499094
8       499047
9       530503
10      499028
11      527034
12      498941
13      498967
14      522071
15      509045
16      499081
17      520208

- 18 sections in organization 360421

In [16]:
dfFlt.filter( F.col('section_id') == 499028).select(*cats['learnerVars']).distinct().toPandas()

learner_id
0      1259558
1      1248382
2      1248826
3      1249701
4      1248402
5      1248370
6      1257687
7      1248404
8      1248792
9      1248400
10     1248786
11     1249343
12     1248378
13     1248403
14     1248371
15     1248369
16     1249244
17     1248377
18     1251888
19     1255484
20     1248372
21     1248788
22     1249507
23     1259171
24     1249243
25     1258474

- 25 learners in section 499028

In [17]:
dfFlt.filter( F.col('section_id') == 499028).select(*cats['assessmentVars']).distinct().toPandas()

assessment_id  assessment_instance_id
0         748408                  799468
1         747201                  926453
2         748403                  799457
3         750551                  820432
4         747722                  799442
5         748402                  907587
6         751391                  842344
7         748402                  907549
8         750014                  923506
9         750043                  799474

- 9 assessments in section 499028
- one instance per assessment per section

###### Display Assignments for Section 498941

In [12]:
dfFlt.filter( F.col('section_id') == 499028).select('assessment_id','assessment_instance_id', *cats['assignmentVars']).distinct().orderBy('assignment_due_date').toPandas()

assessment_id  assessment_instance_id  assessment_instance_attempt_id  \
0           750043                  799474                       1150375.0   
1           750043                  799474                       1188946.0   
2           750043                  799474                       1185322.0   
3           750043                  799474                             NaN   
4           750043                  799474                       1200065.0   
..             ...                     ...                             ...   
138         747201                  926453                       1114924.0   
139         747201                  926453                       1170866.0   
140         747201                  926453                       1099537.0   
141         747201                  926453                       1219376.0   
142         747201                  926453                       1099538.0   

    assignment_due_date assignment_final_submission_date  \
0   2019-10-31 04:59:00              2019-10-31 04:59:00   
1   2019-10-31 04:59:00              2019-10-31 04:59:00   
2   2019-10-31 04:59:00              2019-10-31 04:59:00   
3   2019-10-31 04:59:00              2019-10-31 04:59:00   
4   2019-10-31 04:59:00              2019-10-31 04:59:00   
..                  ...                              ...   
138 2020-03-11 04:59:00              2020-03-11 04:59:00   
139 2020-03-11 04:59:00              2020-03-11 04:59:00   
140 2020-03-11 04:59:00              2020-03-11 04:59:00   
141 2020-03-11 04:59:00              2020-03-11 04:59:00   
142 2020-03-11 04:59:00              2020-03-11 04:59:00   

     assignment_late_submission  assignment_max_attempts  \
0                         False                        1   
1                         False                        1   
2                         False                        1   
3                         False                        1   
4                         False                        1   
..                          ...                      ...   
138                       False                        1   
139                       False                        1   
140                       False                        1   
141                       False                        1   
142                       False                        1   

    assignment_start_date ced_assignment_type_code  is_affecting_grade  \
0     2019-10-24 05:00:00               assessment                True   
1     2019-10-24 05:00:00               assessment                True   
2     2019-10-24 05:00:00               assessment                True   
3     2019-10-24 05:00:00               assessment                True   
4     2019-10-24 05:00:00               assessment                True   
..                    ...                      ...                 ...   
138   2020-03-09 05:00:00               assessment                True   
139   2020-03-09 05:00:00               assessment                True   
140   2020-03-09 05:00:00               assessment                True   
141   2020-03-09 05:00:00               assessment                True   
142   2020-03-09 05:00:00               assessment                True   

    max_student_stop_datetime min_student_start_datetime  \
0     2019-10-28 16:24:37.517    2019-10-24 16:36:52.673   
1     2019-10-28 16:24:37.517    2019-10-24 16:36:52.673   
2     2019-10-28 16:24:37.517    2019-10-24 16:36:52.673   
3     2019-10-28 16:24:37.517    2019-10-24 16:36:52.673   
4     2019-10-28 16:24:37.517    2019-10-24 16:36:52.673   
..                        ...                        ...   
138   2020-03-11 05:00:10.856    2020-03-09 18:14:20.612   
139   2020-03-11 05:00:10.856    2020-03-09 18:14:20.612   
140   2020-03-11 05:00:10.856    2020-03-09 18:14:20.612   
141   2020-03-11 05:00:10.856    2020-03-09 18:14:20.612   
142   2020-03-11 05:00:10.856    2020-03-09 18:14:20.612   

     number_of_d

- 9 assignments (one for each assessment) in section 499028

###### Display Assignment Items for Assessment 748408 of Section 499028

In [11]:
dfFlt.filter( F.col('assessment_id') == 748408).select(*cats['itemVars']).distinct().toPandas()

item_is_offline_scored item_type_code_name scoring_type_code
0                   False               essay            manual
1                   False         shortAnswer            manual
2                   False      multipleChoice         automatic
3                   False            matching         automatic
4                   False               essay      [unassigned]
5                   False           trueFalse         automatic

- 5 items in assessment 721235
- 1 item (true/false) is automatic scored
- 2 item essay is one manual scored the other [unassigned]
- 1 item short answer is manual scored
- 1 item matching is automatic scored

###### Display Assignment Attempts for Assessment 748408

In [9]:
dfFlt.filter( F.col('assessment_id') == 748408 ).select('learner_assignment_attempt_id', *cats['assignmentAttemptVars']).distinct().toPandas()

learner_assignment_attempt_id  assessment_instance_attempt_id  \
0                         1002687                         1231041   
1                         1002584                         1148290   
2                         1002575                         1110794   
3                         1002702                         1227978   
4                         1002648                         1155442   
5                         1002598                         1098979   
6                         1002614                         1122674   
7                         1002831                         1175820   
8                         1002637                         1117904   
9                         1002558                         1190008   
10                        1002669                         1165346   
11                        1002763                         1098986   
12                        1002798                         1125030   
13                        1002807                         1239925   
14                        1002587                         1098981   
15                        1002681                         1193414   
16                        1002814                         1210783   

    assignment_attempt_number  final_score_unweighted  is_force_scored  \
0                           1                      38            False   
1                           1                      44            False   
2                           1                      23            False   
3                           1                      32            False   
4                           1                      46            False   
5                           1                      39            False   
6                           1                      40            False   
7                           1                      44            False   
8                           1                      43            False   
9                           1                      44            False   
10                          1                      20            False   
11                          1                      41            False   
12                          1                      39            False   
13                          1                      45            False   
14                          1                      37            False   
15                          1                      11            False   
16                          1                      29            False   

    is_manual_scoring_required  learner_assignment_attempt_id  \
0                        False                        1002687   
1                        False                        1002584   
2                        False                        1002575   
3                        False                        1002702   
4                        False                        1002648   
5                        False                        1002598   
6                        False                        1002614   
7                        False                        1002831   
8                        False                        1002637   
9                        False                        1002558   
10                       False                        1002669   
11                       False                        1002763   
12                       False                        1002798   
13                       False                        1002807   
14                       False                        1002587   
15                       False                        1002681   
16                       False                        1002814   

   learner_attempt_status  points_possible_unweighted     scored_datetime  \
0            fully scored                        48.0 2019-12-13 10:21:57   
1            fully scored                        48.0 2019-12-13 10:17:50   
2            fully scored 

- 16 attempts
- Majority of students started (student_start_datetime) and stopped (student_stop_datetime) on 2019-12-13
- Scored (was_fully_scored_datetime) between 12/12 and 1/10

###### Display Assignment Attempt Items for Attempt 1098979

In [8]:
dfFlt.filter( F.col('assessment_instance_attempt_id') == 1098979 ).select('learner_assigned_item_attempt_id', *cats['itemAttemptVars']).distinct().toPandas()

learner_assigned_item_attempt_id  assessment_item_response_id  \
0                            1526587                      1548819   
1                            1526589                      1645778   
2                            1526598                      1614874   
3                            1438642                      1648977   
4                            1526590                      1589534   
5                            1526586                      1657844   
6                            1526592                      1572928   
7                            1526585                      1554352   
8                            1526596                      1622899   
9                            1526593                      1570752   
10                           1526581                      1650527   
11                           1526582                      1619280   
12                           1526584                      1599166   
13                           1526597                      1626774   
14                           1526591                      1635215   
15                           1526594                      1569829   
16                           1526583                      1627583   
17                           1526595                      1577219   
18                           1526588                      1600362   

   assigned_item_status  is_deleted  learner_assigned_item_attempt_id  \
0                scored       False                           1526587   
1                scored       False                           1526589   
2                scored       False                           1526598   
3                scored       False                           1438642   
4                scored       False                           1526590   
5                scored       False                           1526586   
6                scored       False                           1526592   
7                scored       False                           1526585   
8                scored       False                           1526596   
9                scored       False                           1526593   
10               scored       False                           1526581   
11               scored       False                           1526582   
12               scored       False                           1526584   
13               scored       False                           1526597   
14               scored       False                           1526591   
15               scored       False                           1526594   
16               scored       False                           1526583   
17               scored       False                           1526595   
18               scored       False                           1526588   

   response_correctness  
0               correct  
1               correct  
2               correct  
3               correct  
4               correct  
5               correct  
6               correct  
7               correct  
8     partially_correct  
9               correct  
10              correct  
11            incorrect  
12              correct  
13              correct  
14            incorrect  
15            incorrect  
16              correct  
17              correct  
18              correct

- 18 items
- All answered (scored)
  - 14 correct
  - 3 incorrect
  - on partially correct


###### Display Assignment Attempt Items for Attempt 1193414

In [22]:
dfFlt.filter( F.col('assessment_instance_attempt_id') == 1193414 ).select(*cats['itemAttemptVars']).distinct().toPandas()

assessment_item_response_id assigned_item_status  is_deleted  \
0                       1662814               scored       False   
1                       1556132               scored       False   
2                       1642668               scored       False   
3                       1575456               scored       False   
4                       1624295               scored       False   
5                       1563437               scored       False   
6                       1552133               scored       False   
7                       1609927               scored       False   
8                       1631174               scored       False   
9                       1583454               scored       False   
10                      1555799               scored       False   
11                      1654912               scored       False   
12                      1559832               scored       False   
13                      1635745               scored       False   
14                      1610358               scored       False   
15                      1623177               scored       False   
16                      1607672               scored       False   
17                      1617860               scored       False   
18                      1567338               scored       False   

    learner_assigned_item_attempt_id response_correctness  
0                            1400105              correct  
1                            1400108            incorrect  
2                            1400094            incorrect  
3                            1400110            incorrect  
4                            1400101            incorrect  
5                            1400114            incorrect  
6                            1400096              correct  
7                            1400099              correct  
8                            1400095            incorrect  
9                            1400104            incorrect  
10                           1400097            incorrect  
11                           1400100            incorrect  
12                           1400111              correct  
13                           1400109              correct  
14                           1400112              correct  
15                           1400103            incorrect  
16                           1400098            incorrect  
17                           1400113              correct  
18                           1400102            incorrect

- 18 items
- All answered (scored)
  - 7 correct
  - 11 incorrect
